compiler/p/codegen/OMRCodeGenerator.cpp

```diff
@@ -1793,6 +1793,11 @@ bool OMR::Power::CodeGenerator::getSupportsOpCodeForAutoSIMD(TR::CPU *cpu, TR::I
         case TR::m2v:
             // only P9 has splat byte immediate, otherwise it's too expensive
             return cpu->isAtLeast(OMR_PROCESSOR_PPC_P9);
+        case TR::s2m:
+            if (et == TR::Int8 || et == TR::Int16 || et == TR::Int32 || et == TR::Int64)
```

Comment:
I think we can assert `et == TR::Int64` here. This is because short has two bytes, meaning mask will have two elements and there they would be Int64 for 128-bit vector.